plugins/experimental/access_control/utils.cc

```diff
@@ -217,32 +209,10 @@ cryptoMessageDigestGet(const char *digestType, const char *data, size_t dataLen,
     return 0;
   }
 
-#ifndef HAVE_HMAC_CTX_NEW
-  HMAC_CTX_init(ctx);
-#else
-  ctx = HMAC_CTX_new();
-#endif
-  if (!HMAC_Init_ex(ctx, key, keyLen, md, nullptr)) {
-    AccessControlError("failed to create EVP message digest context: %s", cryptoErrStr(buffer, sizeof(buffer)));
-    goto err;
-  }
-
-  if (!HMAC_Update(ctx, reinterpret_cast<const unsigned char *>(data), dataLen)) {
-    AccessControlError("failed to update the signing hash: %s", cryptoErrStr(buffer, sizeof(buffer)));
-    goto err;
+  if (!HMAC(md, reinterpret_cast<const unsigned char *>(key), keyLen, reinterpret_cast<const unsigned char *>(data), dataLen,
+            reinterpret_cast<unsigned char *>(out), &len)) {
+    AccessControlError("failed to get the signing hash: %s", cryptoErrStr(buffer, sizeof(buffer)));
```

Comment:
Nice change. Since we have all the data here in one place, no need to use the `HMAC_Update` funcionality.